dali/pipeline/workspace/workspace.h

```diff
@@ -203,6 +203,28 @@ class WorkspaceBase : public ArgumentWorkspace {
                                      StorageDevice::GPU);
   }
 
+
+  /**
+   * @brief Returns true if this workspace has CUDA stream available
+   */
+  virtual bool has_stream() const = 0;
+
+
+  /**
+   * @brief Returns the CUDA stream that this work is to be done in.
+   */
+  cudaStream_t stream() const {
+    DALI_ENFORCE(has_stream(),
+                 "Provided workspace doesn't allow for CUDA calculations. "
```

Comment:
The error message is misleading, I think you can have a device workspace without the stream assigned. 
Maybe keep the old message of: 
```suggestion
                 "Workspace does not have a valid stream."
```
Maybe additional function stating if that workspace supports streams at all?